test/decl/protocol/req/associated_type_ambiguity.swift

```diff
@@ -48,7 +48,7 @@ extension P2 where Self : P3, T == Int {
 }
 
 extension P2 where Self : P3 {
-  func takeT1(_: T) {}
+  func takeT1(_: T) {} // expected-error {{'T' is ambiguous for type lookup in this context}}
```

Comment:
I feel like we've tripped over this specific bug in a number of places, where unqualified lookup is ambiguous but shouldn't be. I suspect your change isn't making it much worse in practice.

Reply:
Sounds good! I'm going to work on other diagnostic...